org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaDiagnosticsTest.java

```diff
@@ -197,6 +197,28 @@ public void cvc_complex_type_3_2_2() throws Exception {
 		testCodeActionsFor(xml, d, ca(d, te(2, 7, 2, 14, "")));
 	}
 
+	@Test
+	public void cvc_complex_type_2_4_bCodeAction_basic() throws Exception {
+		String xml = "<root xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"    xsi:noNamespaceSchemaLocation=\"src/test/resources/xsd/unique.xsd\">\r\n" + //
+				"</root>";
+		Diagnostic d = d(0, 1, 0, 5, XMLSchemaErrorCode.cvc_complex_type_2_4_b);
+		testDiagnosticsFor(xml, d);
+		testCodeActionsFor(xml, d, ca(d, te(2, 0, 2, 0, "\r\n <authors>\r\n  <author></author>\r\n </authors>")));

```

Comment:
The testCodeActionsFor that you are using use tabs instead of spaces https://github.com/eclipse/lemminx/blob/9e879e3de0c363e12651f3aa0cdc82a050d79143/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/XMLAssert.java#L644 So your test must use tabs instead of spaces:

```java
testCodeActionsFor(xml, d, ca(d, te(2, 0, 2, 0, //
				"\r\n" + //
						"\t<authors>\r\n" + //
						"\t\t<author></author>\r\n" + //
						"\t</authors>")));
```